gmeta/src/lib.rs

```diff
@@ -1,66 +1,189 @@
+//! Crate for providing metadata for smart contracts.
+//!
+//! Metadata is used to describe the interface of a smart contract. For example,
+//! it can be used when uploading a contract using <https://idea.gear-tech.io>.
+//! The metadata informs the user about the contract's interface and allows them
+//! to interact with it using custom types.
+//!
+//! To generate a metadata output file `contract_name.meta.txt` for a contract,
+//! you need:
+//!
+//! - Add `gmeta` crate to your `Cargo.toml` file.
+//! - Define an empty struct that will identify the contract metadata.
+//! - Implement the [`Metadata`] trait for this struct by defining the
+//!   associated types of the trait.
+//! - Call [`gear_wasm_builder::build_with_metadata`](https://docs.gear.rs/gear_wasm_builder/fn.build_with_metadata.html)
```

Comment:
we really need crate dep for such refs?

Reply:
`dev-dependencies` are used to compile examples (aka doc-tests).